### PR TITLE
try_cast documentation first draft

### PR DESCRIPTION
**Pull request by celaus**
updated documentation to include try_cast

es upstream fix

### CODE REVIEW

**Comment by kovrus** [comment]
The case when TRY_CAST fails with an error if the the cast is explicitly not permitted is not documented

File: docs/sql/data_types.txt [hunk 1]
```diff
@@ -464,16 +467,78 @@ types, e.g. converting an array of integer values to a boolean array.
...
     +---------------------+
-    | [false, true, true] | 
+    | [false, true, true] |
     +---------------------+
     SELECT 1 row in set (... sec)
 
 .. note::
 
-   It is not possible to cast to or from ``object``, ``geopoint`` and
+   It is not possible to cast to or from ``object``, ``geopoint``, and
```
Comment (mfelsche) [inline docs/sql/data_types.txt]: remove comma

File: docs/sql/data_types.txt [hunk 2]
```diff
@@ -464,16 +467,78 @@ types, e.g. converting an array of integer values to a boolean array.
...
+
+::
+
+   try_cast(expression as type)
+
+
+Example usages:
+::
+
+    cr> select try_cast(name as boolean) from sys.nodes limit 1;
```
Comment (mfelsche) [inline docs/sql/data_types.txt]: ```name``` here, ```port['http']``` in the result table?

File: docs/sql/data_types.txt [hunk 3]
```diff
@@ -464,16 +467,78 @@ types, e.g. converting an array of integer values to a boolean array.
...
+    +-----------------------------------+
+    | TRY_CAST(port['http'] AS boolean) |
+    +-----------------------------------+
+    | TRUE                              |
+    +-----------------------------------+
+    SELECT 1 row in set (... sec)
+
+Same as ``cast``, an integer can be converted to boolean. Trying to cast a
+``string`` to ``boolean`` however, will fail with ``cast`` and return ``null`` with
```
Comment (mfelsche) [inline docs/sql/data_types.txt]: string to boolean can succeed if it is 't', 'f', 'true' or 'false'.

File: docs/sql/data_types.txt [hunk 4]
```diff
@@ -464,16 +467,78 @@ types, e.g. converting an array of integer values to a boolean array.
...
+``try_cast``:
+::
+
+    cr> select try_cast(name as integer) from sys.nodes limit 1;
```
Comment (mfelsche) [inline docs/sql/data_types.txt]: did you mean ```as boolean``` here?

File: docs/sql/data_types.txt [hunk 5]
```diff
@@ -464,16 +467,78 @@ types, e.g. converting an array of integer values to a boolean array.
...
+    SELECT 1 row in set (... sec)
+
+This allows to ignore invalid conversions within aggregate functions and
+returns more homogenious results. For example to find valid IP addresses in a
+column that also stores hostnames:
+::
+
+    cr> select cast(nameOrIp as ip) from hosts limit 100;
+    +--------------------------+
+    | TRY_CAST(port AS string) |
```
Comment (mfelsche) [inline docs/sql/data_types.txt]: different name here as well

File: docs/sql/data_types.txt [hunk 6]
```diff
@@ -464,16 +467,83 @@ types, e.g. converting an array of integer values to a boolean array.
...
+   try_cast(expression as type)
+
+
+Example usages:
+
+::
+
+    cr> select try_cast('true' as boolean) from sys.nodes limit 1;
+    +-----------------------------------+
+    | TRY_CAST('true' AS boolean)         |
```
Comment (mfelsche) [inline docs/sql/data_types.txt]: too many whitespaces

**Comment by mfelsche** [comment]
othwerwise ack

**Pull request merged**